kornia/geometry/camera/distortion_kannala_brandt.py

```diff
@@ -0,0 +1,245 @@
+    k2 = params[..., 6]
+    k3 = params[..., 7]
+
+    radius_sq = x**2 + y**2
+
+    radius = radius_sq.sqrt()
+    radius_inverse = 1.0 / radius
+    theta = radius.atan2(ops.ones_like(radius))
+    theta2 = theta**2
+    theta4 = theta2**2
+    theta6 = theta2 * theta4
+    theta8 = theta4**2
+
+    r_distorted = theta * (1.0 + k0 * theta2 + k1 * theta4 + k2 * theta6 + k3 * theta8)
+
+    scaling = r_distorted * radius_inverse
+
+    u = fx * scaling * x + cx
+    v = fy * scaling * y + cy
+
+    return ops.stack([u, v], dim=-1)
+
+
+def distort_points_kannala_brandt(projected_points_in_camera_z1_plane: Tensor, params: Tensor) -> Tensor:
+    r"""Distort one or more points from the canonical z=1 plane into the camera frame using the Kannala-Brandt
+    model.
+
+    Args:
+        projected_points_in_camera_z1_plane: Tensor representing the points to distort with shape (..., 2).
+        params: Tensor representing the parameters of the Kannala-Brandt distortion model with shape (..., 8).
+
+    Returns:
+        Tensor representing the distorted points with shape (..., 2).
+
+    Example:
+        >>> points = torch.tensor([319.5, 239.5])  # center of a 640x480 image
+        >>> params = torch.tensor([1000.0, 1000.0, 320.0, 280.0, 0.1, 0.01, 0.001, 0.0001])
+        >>> distort_points_kannala_brandt(points, params)
+        tensor([1982.6832, 1526.3619])
+    """
+    KORNIA_CHECK_SHAPE(projected_points_in_camera_z1_plane, ["*", "2"])
+    KORNIA_CHECK_SHAPE(params, ["*", "8"])
+
+    x = projected_points_in_camera_z1_plane[..., 0]
+    y = projected_points_in_camera_z1_plane[..., 1]
+
+    radius_sq = x**2 + y**2
+
+    # TODO: we can optimize this by passing the radius_sq to the impl functions. Check if it's worth it.
+    distorted_points = ops.where(
+        radius_sq[..., None] > 1e-8,
+        _distort_points_kannala_brandt_impl(
+            projected_points_in_camera_z1_plane,
+            params,
+        ),
+        distort_points_affine(projected_points_in_camera_z1_plane, params[..., :4]),
+    )
+
+    return distorted_points
+
+
+def undistort_points_kannala_brandt(distorted_points_in_camera: Tensor, params: Tensor) -> Tensor:
+    r"""Undistort one or more points from the camera frame into the canonical z=1 plane using the Kannala-Brandt
+    model.
+
+    Args:
+        distorted_points_in_camera: Tensor representing the points to undistort with shape (..., 2).
+        params: Tensor representing the parameters of the Kannala-Brandt distortion model with shape (..., 8).
+
+    Returns:
+        Tensor representing the undistorted points with shape (..., 2).
+
+    Example:
+        >>> points = torch.tensor([319.5, 239.5])  # center of a 640x480 image
+        >>> params = torch.tensor([1000.0, 1000.0, 320.0, 280.0, 0.1, 0.01, 0.001, 0.0001])
+        >>> undistort_points_kannala_brandt(points, params).shape
+        torch.Size([2])
+    """
+    KORNIA_CHECK_SHAPE(distorted_points_in_camera, ["*", "2"])
```

Comment:
the `*` is working on check shape now? 🤔

Reply:
i assumed so